src/lib/viewers/media/DashViewer.js

```diff
@@ -1015,6 +1062,54 @@ class DashViewer extends VideoBaseViewer {
         this.renderUI();
     }
 
+    /**
+     * Updates the selected subtitle and updates the player accordingly
+     * @param {string} subtitle - ID of the subtitle track
+     * @emits subtitlechange
+     * @return {void}
+     */
+    setSubtitle(subtitle) {
+        const subtitleIdx = this.textTracks.findIndex(({ id }) => id === subtitle);
+
+        this.cache.set('media-subtitles', subtitle, true);
+        this.cachedSubtitle = this.selectedSubtitle;
+        this.selectedSubtitle = subtitle;
+
+        // Auto-generated index 0 ==> turn auto-generated text track on
+        if (this.autoCaptionDisplayer && subtitleIdx === 0) {
+            // Manually set text visibility with the custom Shaka Text Displayer
+            this.autoCaptionDisplayer.setTextVisibility(true);
+            this.emit('subtitlechange', __('auto_generated'));
+
+            // Valid non-auto-generated index ==> turn specified text track on
+        } else if (this.textTracks[subtitleIdx] !== undefined) {
+            const track = this.textTracks[subtitleIdx];
+            this.player.selectTextTrack(track);
+            this.player.setTextTrackVisibility(true);
+            this.emit('subtitlechange', track.language);
+
+            // Index -1 ==> turn subtitles/captions off
+        } else {
+            if (this.autoCaptionDisplayer) {
+                this.autoCaptionDisplayer.setTextVisibility(false);
+            }
+
+            this.player.setTextTrackVisibility(false);
+            this.emit('subtitlechange', null);
+        }
+
+        this.renderUI();
+    }
+
+    /**
+     * Toggles the subtitles on or off
+     * @param {boolean} showSubtitles - Boolean indicating whether to hide or show the subtitles
+     * @return {void}
+     */
+    toggleSubtitles(showSubtitles) {
+        this.setSubtitle(showSubtitles ? this.cachedSubtitle : SUBTITLES_OFF);
```

Comment:
Can we avoid this third instance property? Can we use `this.selectedSubtitle` or the value in `this.cache` here, instead?

Reply:
The problem with using `selectedSubtitle` or the value in `this.cache` is that those will be updated to `-1` as well to indicate that subtitles are turned off. I could perhaps skip the storing of `-1` to cache and re-use that here